extends Area3D

@export var item: itens # item que est치 sendo adquirido
var area: bool = false # controlado pelos sinais no final do c칩digo (entrou/saiu da 치rea)

@onready var pers: CharacterBody3D = get_tree().get_root().get_node("main/pers")
@onready var pers_label: Label3D = get_tree().get_root().get_node("main/pers/utilidades/Label3D")
@onready var inventarioUI: VBoxContainer = get_tree().get_root().get_node("main/pers/UI/invcontainer/Invent치rio [TAB]")
@onready var UI: Control = get_tree().get_root().get_node("main/pers/UI")

func _ready() -> void:
	if item != null:
		$MeshInstance3D.mesh = item.mesh
		
		
func _process(_delta: float) -> void:
	if !(pers.interagir and area):
		return

	if pers.adicionar_item(item.duplicate(true)):
		UI.atualizarhudUI()
		UI.atualizarinventarioUI()
		pers_label.text = str("")
		queue_free()
	else:
		$PopupPanel.position = $PopupPanel.get_global_mouse_position() - Vector2(65, 0)
		$PopupPanel/erro.popup()




func _on_body_entered(body: Node3D) -> void:
	if body.is_in_group("Player"):
		pers_label.text = str("Pressione E para interagir")
		area = true
		

func _on_body_exited(body: Node3D) -> void:
	if body.is_in_group("Player"):
		pers_label.text = str("")
		area = false
